// Copyright 2020 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package private_statistics.prio.proto;

import "prio/proto/algorithm_parameters.proto";

option java_multiple_files = true;

// Represents the part of the final sum computed by an aggregation server to be
// sent to the destination server which will compute the result.
message PrioSumPart {
  // Universal unique identifier of the PrioDataShareBatch to which batch
  // corresponds to.
  optional string batch_uuid = 1;
  // The value of the sum computed by the aggregation server.
  optional bytes value_sum = 2;
  // Parameters for this Prio aggregation.
  optional PrioAlgorithmParameters parameters = 3;
  // A name for this specific aggregation.
  optional string name = 4;
  // Signature of data.
  optional bytes signature_of_value_sum = 5;
  // Some identifier as to which key was used to sign
  optional string signature_id = 6;
}
